code/go/0chain.net/smartcontract/storagesc/curator.go

```diff
@@ -115,3 +130,19 @@ func (sa StorageAllocation) isCurator(id string) bool {
 	}
 	return false
 }
+
+func curatorToCuratorEvent(ci *curatorInput) *event.Curator {
+	return &event.Curator{
+		AllocationID: ci.AllocationId,
+		CuratorID:    ci.CuratorId,
+	}
+}
+
+func emitCuratorEvent(ci *curatorInput, balances chainstate.StateContextI, txn *transaction.Transaction, eventTag event.EventTag) error {
+	data, err := json.Marshal(curatorToCuratorEvent(ci))
+	if err != nil {
+		return fmt.Errorf("failed to marshal curator: %v", err)
+	}
+	balances.EmitEvent(event.TypeStats, eventTag, txn.Hash, string(data))
```

Comment:
txn_hash is not stored in data, that shouldn't be index
here ci.AllocationID should be 3rd argument index